### PR TITLE
[Memberships] Retrieve valid plans based on a specific tier ID

### PR DESCRIPTION
**Pull request by millerf**
See D145591-code

## Proposed changes:

* See D145591-code

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to '..'
*

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-abstract-token-subscription-service.php [hunk 1]
```diff
@@ -247,6 +248,93 @@ private function maybe_gate_access_for_user_if_post_tier( $post_id, $user_abbrev
...
+			// There is an issue with the meta
+			return new WP_Error( 'wrong-data-plan-not-found', 'The plan related to the tier is missing data' );
+		}
+
+		$valid_plan_ids[] = $tier_id;
+
+		$tier_price = floatval( $tier_price );
+
+		// At this point we know the post is
+		$annual_tier = null;
```
Comment (chrismccluskey) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-abstract-token-subscription-service.php:293]: We might want to define `$annual_tier_price = 0` here.

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-abstract-token-subscription-service.php [hunk 2]
```diff
@@ -247,6 +248,93 @@ private function maybe_gate_access_for_user_if_post_tier( $post_id, $user_abbrev
...
+
+			$plan_price = floatval( $plan_price );
+
+			if ( $tier_currency !== $plan_currency ) {
+				// For now, we don't count if there are different currency (not sure how to convert price in a pure JP env)
+				continue;
+			}
+
+			if ( ( $plan_interval === '1 month' && $plan_price >= $tier_price ) ||
+				( $plan_interval === '1 year' && $plan_price >= $annual_tier_price )
```
Comment (chrismccluskey) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-abstract-token-subscription-service.php:328]: We might also want to make sure that the `$annual_tier_price > 0` here as well, to make sure it got redefined to a real value.

**Review by chrismccluskey — COMMENTED** [review]
I got some errors while testing, but I think that's due to a conflicting function in the WPCOM PR. "An unknown error happened while checking the site plans".

**Comment by millerf** [comment]
Closing in favor of https://github.com/Automattic/jetpack/pull/37084/files

**Pull request closed without merging**